File: c/2.c
#include <stdio.h>

unsigned long long fib(int n){
   if(n <= 3) return n;
   else return fib(n-1) + fib(n-2);
}
int main(){
   int n = 1;
   unsigned long long curr = fib(n);
   unsigned long long sum = 0LL;
   do{
      if( (curr % 2LL) == 0LL){
         sum += curr;
      }
      n++;
      curr = fib(n);
   } while (curr < 4000000LL);

   printf("sum: %lld\n", sum);
   return 0;
}

File: python/2.py
#!/usr/bin/python

#fibonacci with dynamic programming. this would be much more tedious in C
#unless I use the naive method without dp

fibdict={}

def fib(n):
    answer = 0
    if n == 1 or n == 2:
        answer = n
    elif n in fibdict.keys():
        answer = fibdict[n]
    else:
        answer  = fib(n-1) + fib(n-2)
        fibdict[n] = answer
    return answer

def main():
    sum = 0
    n = 1
    curr = fib(n)
    while curr < 4e6:
        # print n, ":", curr
        if curr % 2 == 0:
            sum += curr
        n += 1
        curr = fib(n)
    print sum

if __name__ == "__main__":
    main()
